app/controllers/mixins/ems_common_angular.rb

```diff
@@ -420,6 +423,12 @@ def set_ems_record_vars(ems, mode = nil)
       build_connection(ems, endpoints, mode)
     end
 
+    def get_hostname_from_routes(ems, hostname, port, token)
+      client = ems.class.raw_connect(hostname, port, :bearer => token)
+      routes = client.get_routes
```

Comment:
please change to something specific to only get the ones route that you need:
```
rotues = client.get_route(namespace='openshift-infra', name='hawkular-metrics')
```
and then you won't need to filter that later.